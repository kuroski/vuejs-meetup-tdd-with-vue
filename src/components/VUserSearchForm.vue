<script>
export default {
  name: 'user-search-form',
  props: {
    search: {
      type: Function,
      required: true,
    },
  },
  data() {
    return {
      term: '',
    }
  },
}
</script>

<template>
  <form @submit.prevent="search(term)">
    <el-input
      placeholder="Pesquise o usuÃ¡rio"
      v-model="term"
      class="input-with-select"
    >
      <el-button
        class="submit"
        native-type="submit"
        slot="append"
        icon="el-icon-search"></el-button>
    </el-input>
  </form>
</template>
