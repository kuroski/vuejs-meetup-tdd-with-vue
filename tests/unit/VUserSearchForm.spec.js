import { shallowMount, createLocalVue } from '@vue/test-utils'
import ElementUI from 'element-ui'
import VUserSearchForm from '@/components/VUserSearchForm'

let localVue = createLocalVue()
localVue.use(ElementUI)

describe('VUserSearchForm', () => {
  let props

  const build = () => {
    const wrapper = shallowMount(VUserSearchForm, {
      propsData: props,
      localVue,
    })
    return {
      wrapper,
      input: () => wrapper.find('.input-with-select'),
      button: () => wrapper.find('.submit'),
    }
  }

  beforeEach(() => {
    props = {
      search: jest.fn()
    }
  })

  it('renders the component', () => {
    // arranje
    const { wrapper } = build()
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main components', () => {
    // arranje
    const { input, button } = build()
    // assert
    expect(input().exists()).toBe(true)
    expect(button().exists()).toBe(true)
  })

  it('calls search function with search term on submit', () => {
    // arranje
    const expectedUser = 'kuroki'
    const { wrapper, button } = build()
    wrapper.setData({
      term: expectedUser
    })

    // act
    button().trigger('click')
    button().trigger('submit')
    // assert
    expect(props.search).toHaveBeenCalledWith(expectedUser)
  })
})