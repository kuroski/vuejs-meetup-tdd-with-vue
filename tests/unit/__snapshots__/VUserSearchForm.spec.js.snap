// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserSearchForm renders the component 1`] = `
<form>
  <!---->
</form>
`;
